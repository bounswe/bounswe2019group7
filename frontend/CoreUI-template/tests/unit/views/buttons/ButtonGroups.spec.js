import Vue from 'vue'
import { shallowMount, mount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import ButtonGroups from '@/views/buttons/ButtonGroups'

Vue.use(BootstrapVue)

describe('ButtonGroups.vue', () => {
  it('has a name', () => {
    expect(ButtonGroups.name).toMatch('button-groups')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(ButtonGroups)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is ButtonGroups', () => {
    const wrapper = shallowMount(ButtonGroups)
    expect(wrapper.is(ButtonGroups)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(ButtonGroups)
    expect(wrapper.find('header.card-header').text()).toMatch('Bootstrap button group')
  })
  test('renders correctly', () => {
    const wrapper = mount(ButtonGroups)
    expect(wrapper.element).toMatchSnapshot()
  })
})
