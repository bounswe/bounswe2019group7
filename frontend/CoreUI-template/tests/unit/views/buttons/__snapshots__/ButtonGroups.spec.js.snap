// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroups.vue renders correctly 1`] = `
<div
  class="wrapper"
>
  <div
    class="animated fadeIn"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <div
          class="card"
        >
          <!---->
          <header
            class="card-header"
          >
            <div>
              <i
                class="fa fa-align-justify"
              />
              <strong>
                 Bootstrap button group
              </strong>
               
              <div
                class="card-header-actions"
              >
                <a
                  class="card-header-action"
                  href="https://bootstrap-vue.js.org/docs/components/button-group"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  <small
                    class="text-muted"
                  >
                    docs
                  </small>
                </a>
              </div>
            </div>
          </header>
          <div
            class="card-body"
          >
            <!---->
            <!---->
             
            <div>
              <div
                class="btn-group"
                role="group"
              >
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  One
                </button>
                 
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Two
                </button>
                 
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Three
                </button>
                 
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Four
                </button>
                 
                <button
                  class="btn d-sm-down-none btn-secondary"
                  type="button"
                >
                  Five
                </button>
              </div>
               
              <br />
              <br />
               
              <div
                class="btn-group"
                role="group"
              >
                <button
                  class="btn d-sm-down-none btn-success"
                  type="button"
                >
                  Success
                </button>
                 
                <button
                  class="btn btn-info"
                  type="button"
                >
                  Info
                </button>
                 
                <button
                  class="btn btn-warning"
                  type="button"
                >
                  Warn
                </button>
                 
                <button
                  class="btn d-sm-down-none btn-primary"
                  type="button"
                >
                  Primary
                </button>
                 
                <button
                  class="btn btn-danger"
                  type="button"
                >
                  Danger
                </button>
                 
                <button
                  class="btn btn-link"
                  type="button"
                >
                  Link
                </button>
              </div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
       
      <div
        class="col-12"
      >
        <div
          class="card"
        >
          <!---->
          <header
            class="card-header"
          >
            <div>
              <i
                class="fa fa-align-justify"
              />
              <strong>
                 Button group 
              </strong>
              sizing
          
            </div>
          </header>
          <div
            class="card-body"
          >
            <!---->
            <!---->
             
            <div>
              <div
                class="btn-group"
                role="group"
              >
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Left
                </button>
                 
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Middle
                </button>
                 
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Right
                </button>
              </div>
               
              <br />
              <br />
               
              <div
                class="btn-group btn-group-sm"
                role="group"
              >
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Left
                </button>
                 
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Middle
                </button>
                 
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Right
                </button>
              </div>
               
              <br />
              <br />
               
              <div
                class="btn-group btn-group-lg"
                role="group"
              >
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Left
                </button>
                 
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Middle
                </button>
                 
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Right
                </button>
              </div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
       
      <div
        class="col-12"
      >
        <div
          class="card"
        >
          <!---->
          <header
            class="card-header"
          >
            <div>
              <i
                class="fa fa-align-justify"
              />
              <strong>
                 Button group 
              </strong>
              dropdown support
          
            </div>
          </header>
          <div
            class="card-body"
          >
            <!---->
            <!---->
             
            <div>
              <div
                class="btn-group"
                role="group"
              >
                <button
                  class="btn d-sm-down-none btn-secondary"
                  type="button"
                >
                  Button 1
                </button>
                 
                <button
                  class="btn d-sm-down-none btn-secondary"
                  type="button"
                >
                  Button 2
                </button>
                 
                <div
                  class="dropdown btn-group b-dropdown"
                >
                  <!---->
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="btn dropdown-toggle btn-success"
                    type="button"
                  >
                    Menu
                  </button>
                  <ul
                    class="dropdown-menu dropdown-menu-right"
                    role="menu"
                    tabindex="-1"
                  >
                    <li>
                      <a
                        class="dropdown-item"
                        href="#"
                        role="menuitem"
                        target="_self"
                      >
                        Item 1
                      </a>
                    </li>
                     
                    <li>
                      <a
                        class="dropdown-item"
                        href="#"
                        role="menuitem"
                        target="_self"
                      >
                        Item 2
                      </a>
                    </li>
                     
                    <li>
                      <hr
                        aria-orientation="horizontal"
                        class="dropdown-divider"
                        role="separator"
                      />
                    </li>
                     
                    <li>
                      <a
                        class="dropdown-item"
                        href="#"
                        role="menuitem"
                        target="_self"
                      >
                        Item 3
                      </a>
                    </li>
                  </ul>
                </div>
                 
                <button
                  class="btn d-sm-down-none btn-secondary"
                  type="button"
                >
                  Button 3
                </button>
                 
                <div
                  class="dropdown btn-group b-dropdown"
                >
                  <button
                    class="btn btn-info"
                    type="button"
                  >
                    Split Menu
                  </button>
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="btn dropdown-toggle btn-info dropdown-toggle-split"
                    type="button"
                  >
                    <span
                      class="sr-only"
                    >
                      Toggle Dropdown
                    </span>
                  </button>
                  <ul
                    class="dropdown-menu dropdown-menu-right"
                    role="menu"
                    tabindex="-1"
                  >
                    <li>
                      <a
                        class="dropdown-item"
                        href="#"
                        role="menuitem"
                        target="_self"
                      >
                        Item 1
                      </a>
                    </li>
                     
                    <li>
                      <a
                        class="dropdown-item"
                        href="#"
                        role="menuitem"
                        target="_self"
                      >
                        Item 2
                      </a>
                    </li>
                     
                    <li>
                      <hr
                        aria-orientation="horizontal"
                        class="dropdown-divider"
                        role="separator"
                      />
                    </li>
                     
                    <li>
                      <a
                        class="dropdown-item"
                        href="#"
                        role="menuitem"
                        target="_self"
                      >
                        Item 3
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
       
      <div
        class="col-12"
      >
        <div
          class="card"
        >
          <!---->
          <header
            class="card-header"
          >
            <div>
              <i
                class="fa fa-align-justify"
              />
              <strong>
                 Button group 
              </strong>
              vertical variation
          
            </div>
          </header>
          <div
            class="card-body"
          >
            <!---->
            <!---->
             
            <div>
              <div
                class="btn-group-vertical"
                role="group"
              >
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Top
                </button>
                 
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Middle
                </button>
                 
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Bottom
                </button>
              </div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
       
      <div
        class="col-12"
      >
        <div
          class="card"
        >
          <!---->
          <header
            class="card-header"
          >
            <div>
              <i
                class="fa fa-align-justify"
              />
               
              <strong>
                Button toolbar
              </strong>
               
              <small>
                with button groups
              </small>
               
              <div
                class="card-header-actions"
              >
                <a
                  class="card-header-action"
                  href="https://bootstrap-vue.js.org/docs/components/button-toolbar"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  <small
                    class="text-muted"
                  >
                    docs
                  </small>
                </a>
              </div>
            </div>
          </header>
          <div
            class="card-body"
          >
            <!---->
            <!---->
             
            <div>
              <div
                aria-label="Toolbar with button groups"
                class="btn-toolbar"
                role="toolbar"
                tabindex="0"
              >
                <div
                  class="mx-1 btn-group"
                  role="group"
                >
                  <button
                    class="btn d-sm-down-none btn-secondary"
                    tabindex="-1"
                    type="button"
                  >
                    «
                  </button>
                   
                  <button
                    class="btn btn-secondary"
                    tabindex="-1"
                    type="button"
                  >
                    ‹
                  </button>
                </div>
                 
                <div
                  class="mx-1 btn-group"
                  role="group"
                >
                  <button
                    class="btn d-sm-down-none btn-secondary"
                    tabindex="-1"
                    type="button"
                  >
                    Edit
                  </button>
                   
                  <button
                    class="btn btn-secondary"
                    tabindex="-1"
                    type="button"
                  >
                    Undo
                  </button>
                   
                  <button
                    class="btn btn-secondary"
                    tabindex="-1"
                    type="button"
                  >
                    Redo
                  </button>
                </div>
                 
                <div
                  class="mx-1 btn-group"
                  role="group"
                >
                  <button
                    class="btn btn-secondary"
                    tabindex="-1"
                    type="button"
                  >
                    ›
                  </button>
                   
                  <button
                    class="btn d-sm-down-none btn-secondary"
                    tabindex="-1"
                    type="button"
                  >
                    »
                  </button>
                </div>
              </div>
            </div>
             
            <hr
              class="d-sm-down-none"
            />
             
            <div>
              <div
                aria-label="Toolbar with button groups and input groups"
                class="btn-toolbar d-sm-down-none"
                role="toolbar"
              >
                <div
                  class="mx-1 btn-group btn-group-sm"
                  role="group"
                >
                  <button
                    class="btn btn-secondary"
                    type="button"
                  >
                    New
                  </button>
                   
                  <button
                    class="btn btn-secondary"
                    type="button"
                  >
                    Edit
                  </button>
                </div>
                 
                <div
                  class="input-group w-25 mx-1 input-group-sm"
                  role="group"
                >
                  <!---->
                  <div
                    class="input-group-prepend"
                  >
                    <div
                      class="input-group-text"
                    >
                      $
                    </div>
                  </div>
                   
                  <input
                    class="text-right form-control"
                    type="text"
                  />
                   
                  <div
                    class="input-group-append"
                  >
                    <div
                      class="input-group-text"
                    >
                      .00
                    </div>
                  </div>
                  <!---->
                </div>
                 
                <div
                  class="input-group w-25 mx-1 input-group-sm"
                  role="group"
                >
                  <!---->
                  <div
                    class="input-group-prepend"
                  >
                    <div
                      class="input-group-text"
                    >
                      Size
                    </div>
                  </div>
                   
                  <select
                    class="custom-select"
                  >
                    <option
                      value="Large"
                    >
                      Large
                    </option>
                    <option
                      value="Medium"
                    >
                      Medium
                    </option>
                    <option
                      value="Small"
                    >
                      Small
                    </option>
                  </select>
                  <!---->
                </div>
                 
                <div
                  class="mx-1 btn-group btn-group-sm"
                  role="group"
                >
                  <button
                    class="btn btn-secondary"
                    type="button"
                  >
                    Save
                  </button>
                   
                  <button
                    class="btn btn-secondary"
                    type="button"
                  >
                    Cancel
                  </button>
                </div>
              </div>
            </div>
             
            <hr />
             
            <div>
              <div
                aria-label="Toolbar with button groups and dropdown menu"
                class="btn-toolbar"
                role="toolbar"
              >
                <div
                  class="mx-1 d-sm-down-none btn-group"
                  role="group"
                >
                  <button
                    class="btn btn-secondary"
                    type="button"
                  >
                    New
                  </button>
                   
                  <button
                    class="btn btn-secondary"
                    type="button"
                  >
                    Edit
                  </button>
                   
                  <button
                    class="btn btn-secondary"
                    type="button"
                  >
                    Undo
                  </button>
                </div>
                 
                <div
                  class="dropdown btn-group b-dropdown mx-1"
                >
                  <!---->
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="btn dropdown-toggle btn-secondary"
                    type="button"
                  >
                    menu
                  </button>
                  <ul
                    class="dropdown-menu dropdown-menu-right"
                    role="menu"
                    tabindex="-1"
                  >
                    <li>
                      <a
                        class="dropdown-item"
                        href="#"
                        role="menuitem"
                        target="_self"
                      >
                        Item 1
                      </a>
                    </li>
                     
                    <li>
                      <a
                        class="dropdown-item"
                        href="#"
                        role="menuitem"
                        target="_self"
                      >
                        Item 2
                      </a>
                    </li>
                     
                    <li>
                      <a
                        class="dropdown-item"
                        href="#"
                        role="menuitem"
                        target="_self"
                      >
                        Item 3
                      </a>
                    </li>
                  </ul>
                </div>
                 
                <div
                  class="mx-1 btn-group"
                  role="group"
                >
                  <button
                    class="btn btn-secondary"
                    type="button"
                  >
                    Save
                  </button>
                   
                  <button
                    class="btn btn-secondary"
                    type="button"
                  >
                    Cancel
                  </button>
                </div>
              </div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;
